### PR TITLE
Changes to VSCode Extension

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
* Use the module template feature in the ballerina add comment for providing Ballerina-Integrator templates to users.
* Remove the placeholder functionality which was previously provided. Separate conf files will be used inside template modules for this purpose.
* Remove the default snippet functionality, since it would be provided separately in align with Ballerina snippets.

## Goals
* Allow a user to create a new Ballerina project through the extension.
* Allow a user to create a new Ballerina module using a template.
* Display the list of available Ballerina-Integrator template modules.

### CODE REVIEW

File: vscode/extension/src/createProject.ts [hunk 1]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                projectName = await window.showInputBox({
+                    value: "new_project",
+                    prompt: "Enter value for project name",
+                    placeHolder: "new_project"
+                }).then(text => text);
+                if (projectName != undefined) {
+                    projectPath = await openDialogForFolder();
+                    let projectUri = vscode.Uri.parse(projectPath);
+                    if (projectPath != undefined) {
+                        const cp = require('child_process')
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                        const cp = require('child_process');
```

File: vscode/extension/src/createProject.ts [hunk 2]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                        await cp.exec('cd ' + projectUri.path + ' && ballerina new ' + projectName, (err, stdout, stderr) => {
+                            var message = "Created new ballerina project";
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                            const message = "Created new ballerina project";
```

File: vscode/extension/src/createProject.ts [hunk 3]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                moduleName = await window.showInputBox({
+                    value: templateSelected,
+                    prompt: "Enter value for module name",
+                    placeHolder: templateSelected
+                }).then(text => text);
+                if (moduleName != undefined) {
+                    folderPath = await openDialogForFolder();
+                    let uri = vscode.Uri.parse(folderPath);
+                    if (folderPath != undefined) {
+                        const cp = require('child_process')
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                        const cp = require('child_process');
```

File: vscode/extension/src/createProject.ts [hunk 4]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                        var addCommand = 'cd ' + uri.path + ' && ballerina add ' + moduleName + ' -t wso2/' + templateSelected;
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                        const addCommand = 'cd ' + uri.path + ' && ballerina add ' + moduleName + ' -t wso2/' 
                            + templateSelected;
```

File: vscode/extension/src/createProject.ts [hunk 5]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                        await cp.exec(addCommand, (err, stdout, stderr) => {
+                            if (err) {
+                                window.showErrorMessage("Error: " + err);
+                            } else if (stderr) {
+                                var message = "not a ballerina project";
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                                const message = "not a ballerina project";
```

File: vscode/extension/src/createProject.ts [hunk 6]
```diff
@@ -32,60 +28,90 @@ import { mapToObj } from './utils';
...
+                                var successMessage = "Added new ballerina module";
```
Comment (pramodya1994) [inline vscode/extension/src/createProject.ts]:
```suggestion
                                const successMessage = "Added new ballerina module";
```

File: vscode/extension/src/createProject.ts (hunk 1 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

File: vscode/extension/src/createProject.ts (hunk 2 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

File: vscode/extension/src/createProject.ts (hunk 3 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

File: vscode/extension/src/createProject.ts (hunk 4 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

File: vscode/extension/src/createProject.ts (hunk 5 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

File: vscode/extension/src/createProject.ts (hunk 6 above)
Comment (IrushiL) [reply vscode/extension/src/createProject.ts]: Fixed

**Pull request merged**